File: api/krameriusplus-core-v2/src/main/java/cz/inqool/dl4dh/krameriusplus/corev2/request/enrichment/request/EnrichmentRequest.java
package cz.inqool.dl4dh.krameriusplus.corev2.request.enrichment.request;

import cz.inqool.dl4dh.krameriusplus.corev2.job.KrameriusJobInstance;
import cz.inqool.dl4dh.krameriusplus.corev2.job.config.EnrichmentJobConfig;
import cz.inqool.dl4dh.krameriusplus.corev2.request.Request;
import cz.inqool.dl4dh.krameriusplus.corev2.request.enrichment.item.EnrichmentRequestItem;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "kplus_enrichment_request")
public class EnrichmentRequest extends Request {

    @OneToMany
    @JoinColumn(name = "enrichment_request_id")
    private List<EnrichmentJobConfig> configs = new ArrayList<>();

    @OneToMany
    @JoinColumn(name = "enrichment_request_id")
    private List<EnrichmentRequestItem> items = new ArrayList<>();

    @OneToOne
    private KrameriusJobInstance createRequestJob;
}

File: web/src/models/request/enrichment-request.ts
import { EnrichmentJobConfig } from 'models/job/config/enrichment-job-config'
import { KrameriusJobInstance } from 'models/job/kramerius-job-instance'
import { EnrichmentRequestItem } from './enrichment-request-item'
import { Request } from './request'

export interface EnrichmentRequest extends Request {
	configs: EnrichmentJobConfig[]
	items: EnrichmentRequestItem[]
	createRequestJob: KrameriusJobInstance
}
